presto-hive/src/main/java/com/facebook/presto/hive/parquet/ParquetRecordCursorProvider.java

```diff
@@ -83,7 +80,10 @@ public ParquetRecordCursorProvider(boolean useParquetColumnNames)
         }
 
         // are all columns supported by Parquet code
-        List<HiveColumnHandle> unsupportedColumns = ImmutableList.copyOf(filter(columns, not(isParquetSupportedType())));
+        List<HiveColumnHandle> unsupportedColumns = columns.stream()
+                .filter(columnHandle -> !columnHandle.isPartitionKey())
+                .filter(columnHandle -> columnHandle.getHiveType().equals(HiveType.HIVE_DATE))
+                .collect(toList());
```

Comment:
Yes, `toList()` is ok. At some point we'll move `ImmutableCollectors` to airlift and we can reuse everywhere.